src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.WebSearch/Main.cs

```diff
@@ -0,0 +1,412 @@
+        private void UpdateBrowserIconPath(Theme newTheme)
+        {
+            try
+            {
+                string progId = GetRegistryValue(
+                    "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\http\\UserChoice",
+                    "ProgId");
+
+                if (progId.StartsWith("Opera", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Opera user preferences file:
+                    string prefFile;
+
+                    if (progId.Contains("GX", StringComparison.OrdinalIgnoreCase))
+                    {
+                        _browserName = "Opera GX";
+                        prefFile = System.IO.File.ReadAllText($"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\\Opera Software\\Opera GX Stable\\Preferences");
+                    }
+                    else
+                    {
+                        _browserName = "Opera";
+                        prefFile = System.IO.File.ReadAllText($"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\\Opera Software\\Opera Stable\\Preferences");
+                    }
+
+                    string url = ExtraStringMethods.GetJSONStringValue(prefFile, new string[] { "default_search_provider_data", "template_url_data", "url" });
+
+                    if (url is null)
+                    {
+                        // "default_search_provider_data" doesn't exist in Preferences if the user hasn't searched for the first time,
+                        // therefore we set `url` to opera's default search engine.
+                        url = "https://www.google.com/search?client=opera&q={0}&sourceid=opera";
+                    }
+                    else
+                    {
+                        url = url
+                            .Replace("{searchTerms}", "{0}", StringComparison.Ordinal)
+                            .Replace("{inputEncoding}", "UTF-8", StringComparison.Ordinal)
+                            .Replace("{outputEncoding}", "UTF-8", StringComparison.Ordinal);
+
+                        // just in case there are other url parameters
+                        for (int i = ExtraStringMethods.GetStrNthIndex(url, '}', 2); i != -1; i = ExtraStringMethods.GetStrNthIndex(url, '}', 2))
+                        {
+                            for (int j = i - 1; j > 0; --j)
+                            {
+                                if (url[j] == '&')
+                                {
+                                    url = url.Remove(j, i - j + 1);
+                                    break;
+                                }
+                            }
+                        }
+                    }
+
+                    _searchEngineUrl = url;
+                }
+                else
+                {
+                    if (progId.StartsWith("Chrome", StringComparison.OrdinalIgnoreCase))
```

Comment:
Why do we need this? This feels a bit ugly because we detect predefined browsers.

Reply:
Because `ProgId`'s value starts with the browser's name and then has some (maybe not) random charachters. For example on my PC: `FirefoxURL-308046B0AF4A39CB`.

I couldn't find another way to get the browser's name reliably.

Reply:
But where do we use the browser name and why?

Reply:
In the result's subtitle

Reply:
Could this help: https://docs.microsoft.com/en-us/windows/win32/shell/fa-progids

Can't check at the moment.

Reply:
yes thanks